addons/web/static/src/core/emoji_picker/emoji_picker.js

```diff
@@ -33,7 +32,17 @@ import { useService } from "@web/core/utils/hooks";
  */
 export function useEmojiPicker(ref, props, options = {}) {
     const targets = [];
-    const popover = usePopover(EmojiPicker, { ...options, popoverClass: "o-fast-popover" });
+    let onDropdownCloseCallbacks = [];
+    const popover = usePopover(EmojiPicker, {
+        ...options,
+        popoverClass: "o-fast-popover",
+        onClose: () => {
+            onDropdownCloseCallbacks.forEach(callback => callback());
+            onDropdownCloseCallbacks = [];
+            options.onClose?.();
+        },
+    });
+
```

Comment:
I would rename `onDropdownClose` to `onWillClose`. This is easier to understand the nuance with `onClose` as it matches a bit the wording of OWL component lifecycles